### PR TITLE
Add Update Workout feature

### PR DESCRIPTION
**Pull request by TianaiYan**

### CODE REVIEW

File: src/main/java/commands/WorkoutCommand.java [hunk 1]
```diff
@@ -119,7 +123,8 @@ public void execute() {
         } catch (NumberFormatException e) {
             System.out.println("Uh oh, a number was expected in your input, but a non-formattable\n"
                     + "number was received.");
-            System.out.println("Please try again.");
+            System.out.println("Please try again.Alternatively, type 'help' if you need\n"
```
Comment (Musfirahe0556596) [inline src/main/java/commands/WorkoutCommand.java:126]: Loved that follow up action is added! Just a minor minor formatting issue - add a space in between "again." and "Alternatively"! 👍

File: src/main/java/data/workouts/WorkoutList.java [hunk 2]
```diff
@@ -123,7 +123,32 @@ public Workout deleteWorkout(String userArgument) throws WorkoutOutOfRangeExcept
         return deletedWorkout;
     }
 
-    public boolean checkDeleteIndexIsWithinRange(int indexToDelete) {
-        return indexToDelete > 0 && indexToDelete <= workoutsList.size();
+
+    private boolean checkIndexIsWithinRange(int index) {
```
Comment (Musfirahe0556596) [inline src/main/java/data/workouts/WorkoutList.java:127]: Loved how this is generalized for both update and delete function! Thank you :)!

**Review by Musfirahe0556596 — APPROVED** [review]
Looks good overall! Thank you:) ^_^

**Review by a1021492980 — APPROVED** [review]
Good PR in overall! Keep up the good work!

File: src/main/java/commands/WorkoutCommand.java [hunk 3]
```diff
@@ -95,6 +94,11 @@ public void execute() {
                 getUI().printDeleteWorkoutMessage(deletedWorkout);
                 getFileManager().rewriteAllWorkoutsToFile(getWorkoutList());
                 break;
+            case UPDATE_ACTION_KEYWORD:
+                Workout updateWorkout = getWorkoutList().updateWorkout(getUserArguments());
```
Comment (alanlowzies) [inline src/main/java/commands/WorkoutCommand.java:98]: Just a minor naming suggestion: perhaps `updateWorkout` can be renamed to `updatedWorkout`? To better indicate that this variable will store the `Workout` object that has been updated. 🙂

File: src/main/java/data/workouts/Workout.java [hunk 4]
```diff
@@ -20,6 +20,10 @@ public int getRepetitions() {
         return repetitions;
     }
 
+    public void setRepetitions(int repetitions) {
+        this.repetitions = repetitions;
+    }
```
Comment (alanlowzies) [inline src/main/java/data/workouts/Workout.java:25]: Good use of accessor method to update a private field 🙂👍🏻

File: src/main/java/data/workouts/WorkoutList.java (hunk 2 above)
Comment (alanlowzies) [reply src/main/java/data/workouts/WorkoutList.java:127]: Agreed, helps to minimise duplicated codes 👍🏻

**Review by alanlowzies — COMMENTED** [review]
Implementation of the update feature looks good! Just some code quality and standards for you to consider. Thanks 🙂

File: src/main/java/commands/WorkoutCommand.java (hunk 3 above)
Comment (TianaiYan) [reply src/main/java/commands/WorkoutCommand.java:98]: Thank you for your suggestion.

**Review by alanlowzies — APPROVED** [review]

**Pull request merged**